src/arrow-util/src/builder.rs

```diff
@@ -160,6 +160,29 @@ fn scalar_to_arrow_datatype(scalar_type: &ScalarType) -> Result<DataType, anyhow
             // MZ timestamps are in milliseconds
             DataType::Timestamp(arrow::datatypes::TimeUnit::Millisecond, None)
         }
+        ScalarType::Numeric { max_scale } => {
+            // Arrow/parquet require specifying the precision & scale for the entire column, not
+            // per value, so the decimal point is fixed for all values. Since our numeric type can
```

Comment:
I don't think this is quite right.

From [PostgreSQL](https://www.postgresql.org/docs/current/datatype-numeric.html):

> The precision of a numeric is the total count of significant digits in the whole number, that is, the number of digits to both sides of the decimal point... So the number 23.5141 has a precision of 6 and a scale of 4.

From [MZ](https://materialize.com/docs/sql/types/numeric/#numeric-definitions):

> All `numeric` values have a precision of 39, which cannot be changed

This means the maximum number of digits that we need to represent is 39, not 78.

Reply:
Ah, I read that correctly now. Sorry about that. I'd reframe the comment like:

> Materialize allows 39 digits of precision for numeric values, but allows
> aribtrary scales among those values. e.g. `1e38` and `1e-39` are both valid in
> the same column. However, Arrow/Parquet only allows static declaration of both
> the precision _and_ the scale. To the full range of values of a numeric column,
> we would need 78-digits to store all possible values. Arrow's Decimal256 type
> can only support 76 digis, so we are be unable to represent the entire range.
> 
> To work around this, we try to represent most values in a Decimal128 type which
> can handle 38 digits of precision, using Arrow's default scale of 10 if
> max-scale is not set. Decimal128 provides more compatibility with parquet
> clients than Decimal256. We will error if we encounter a value that is too large
> to represent, and if that happens a user can choose to cast the column to a
> string to represent the value.